### PR TITLE
new crds_s3_get plugin with retries, plus checksum and size check

### PR DESCRIPTION
**Pull request by bhayden53**
* crds_s3_get plugin provided by CRDS devs (eslavich)
* the new plugin includes retries for certain types of failures; both these retries and the ones added into caldp.sysexit are needed to catch different types of CRDS failures which may not throw catchable exceptions in the right places for each package to pick them up and retry. For example, reference file download failures from S3 do not throw an exception that caldp.process can catch, so they can only be retried right in crds_s3_get
* also includes file size and checksum checking, which is valuable based on the several-stage transfers the reference files have to go through to reach a Batch job container
* Since caldp is still in python 3.6, crds in the environment is stuck to 10.1.0, and it could be a while before a change to the plugin makes it to an official astroconda environment; we will leave the plugin directly in caldp until we can pull it in officially.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -40,6 +40,10 @@ RUN mv /etc/ssl/certs/ca-bundle.crt /etc/ssl/certs/ca-bundle.crt.org && \
     mkdir -p /etc/pki/ca-trust/extracted/openssl
 
 USER root
+# this is a temporary solution for an update to the crds_s3_get plugin
```
Comment (jaytmiller) [inline Dockerfile:43]: I'd guess this works and is temporary,  but unless crds_s3_get is needed before caldp is installed,  crds_s3_get will be installed in the conda environment's bin directory so you shouldn't have to do this.

File: scripts/caldp-config-aws [hunk 2]
```diff
@@ -8,7 +8,7 @@ export CRDS_REFERENCE_URI=${CRDSBUCKET}/references/hst/
 export CRDS_PICKLE_URI=${CRDSBUCKET}/pickles/hst/
 export CRDS_SERVER_URL=https://hst-crds-serverless.stsci.edu
 export CRDS_DOWNLOAD_MODE=plugin
-export CRDS_DOWNLOAD_PLUGIN='crds_s3_get ${SOURCE_URL} ${OUTPUT_PATH} ${FILE_SIZE} ${FILE_SHA1SUM}'
+export CRDS_DOWNLOAD_PLUGIN='/home/developer/crds_s3_get ${SOURCE_URL} ${OUTPUT_PATH} --file-size ${FILE_SIZE} --file-sha1sum ${FILE_SHA1SUM}'
```
Comment (jaytmiller) [inline scripts/caldp-config-aws:11]: If you change to relying on the conda install,  remove /home/developer.

File: scripts/crds_s3_get [hunk 3]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
```
Comment (jaytmiller) [inline scripts/crds_s3_get:1]: Make sure this is chmod'ed executable or the conda install won't be executable.

**Review by jaytmiller — APPROVED** [review]
May want to simplify but LGTM.  Note that when CRDS does have crds_s3_get,  I think it will conflict with this one and CALDP's will overwrite the official version since it is installed second.

**Comment by bhayden53** [comment]
> May want to simplify but LGTM. Note that when CRDS does have crds_s3_get, I think it will conflict with this one and CALDP's will overwrite the official version since it is installed second.

I was partly extremely over-cautious here; since we can specify the full path to the plugin we want to use, and I was aware that there was an "official" crds_s3_get floating around somewhere in the container, it seemed safest to me to simply treat this as a "custom" crds plugin and then just be totally explicit. I didn't want to end up in a scenario where we had multiple versions installed in different places and were reliant on what order they showed up in the PATH in order to pick out the right one. 

I also even thought about just naming it something else entirely, but in the end, decided on the "this is very obviously a temporary workaround" approach.

**Comment by jaytmiller** [comment]
OK,  makes sense.

**Pull request merged**